src/webgpu/capability_info.ts

```diff
@@ -38,23 +38,32 @@ export const kQueryTypes = keysOf(kQueryTypeInfo);
 /** Required alignment of a GPUBuffer size, by spec. */
 export const kBufferSizeAlignment = 4;
 
-/** Per-GPUBufferUsage info. */
-export const kBufferUsageInfo: {
-  readonly [k in valueof<typeof GPUConst.BufferUsage>]: {};
+/** Per-GPUBufferUsage copy info. */
+export const kBufferUsageCopyInfo: {
+  readonly [name: string]: GPUBufferUsageFlags;
 } = /* prettier-ignore */ {
-  [GPUConst.BufferUsage.MAP_READ]:      {},
-  [GPUConst.BufferUsage.MAP_WRITE]:     {},
-  [GPUConst.BufferUsage.COPY_SRC]:      {},
-  [GPUConst.BufferUsage.COPY_DST]:      {},
-  [GPUConst.BufferUsage.INDEX]:         {},
-  [GPUConst.BufferUsage.VERTEX]:        {},
-  [GPUConst.BufferUsage.UNIFORM]:       {},
-  [GPUConst.BufferUsage.STORAGE]:       {},
-  [GPUConst.BufferUsage.INDIRECT]:      {},
-  [GPUConst.BufferUsage.QUERY_RESOLVE]: {},
+  'none':     0,
+  'src':      Number(GPUConst.BufferUsage.COPY_SRC),
+  'dst':      Number(GPUConst.BufferUsage.COPY_DST),
+  'src-dest': Number(GPUConst.BufferUsage.COPY_SRC) | Number(GPUConst.BufferUsage.COPY_DST),
 };
+/** List of all GPUBufferUsage copy values. */
+export const kBufferUsageCopy = keysOf(kBufferUsageCopyInfo);
+
+/** Per-GPUBufferUsage keys and info. */
+export const kBufferUsageKeys = Object.keys(GPUConst.BufferUsage);
```

Comment:
the type of this is just `string[]` because Object.keys doesn't do more magic stuff.

So we need `as` to narrow it more. `GPUConst.BufferUsage` is an object, not a type. So `typeof` gives us the type of it:
```typescript
// typeof GPUConst.BufferUsage
type BufferUsageT = {
    readonly MAP_READ: 1;
    readonly MAP_WRITE: 2;
    readonly COPY_SRC: 4;
    readonly COPY_DST: 8;
    readonly INDEX: 16;
    readonly VERTEX: 32;
    readonly UNIFORM: 64;
    readonly STORAGE: 128;
    readonly INDIRECT: 256;
    readonly QUERY_RESOLVE: 512;
}
```
`keyof` will give you a type that is a union of the possible keys of the type.

```typescript
// keyof BufferUsageT
type BufferUsageKeys = "MAP_READ" | "MAP_WRITE" | "COPY_SRC" | "COPY_DST" | "INDEX" | "VERTEX" | "UNIFORM" | "STORAGE" | "INDIRECT" | "QUERY_RESOLVE";
```


```suggestion
type BufferUsageKey = keyof typeof GPUConst.BufferUsage;

export const kBufferUsageKeys = Object.keys(GPUConst.BufferUsage) as BufferUsageKey[];
```

Reply:
Ah... this is so much nicer. Thanks for the suggestions!